nodejs/guestbook/src/backend/app.js

```diff
@@ -0,0 +1,29 @@
+const express = require('express')
+const mongoose = require('mongoose')
+const app = express()
+const routes = require('./routes')
+const PORT = process.env.PORT || 8080;
+
+const MONGO_USERNAME = process.env.MONGO_USERNAME || 'root'
+const MONGO_PASSWORD = process.env.MONGO_PASSWORD || 'password'
+const MONGO_HOST = process.env.MONGO_HOST || 'localhost'
+const MONGO_PORT = process.env.MONGO_PORT || '27017'
+
+// const MONGO_URI = `mongodb://${MONGO_USERNAME}:${MONGO_PASSWORD}@${MONGO_HOST}:${MONGO_PORT}/admin`
+
+const MONGO_URI  = process.env.DB_ADDRESS || 'mongodb://localhost:27017/test';
+
+mongoose.connect(MONGO_URI, {useNewUrlParser: true})
+
+const db = mongoose.connection;
+db.on('error', console.error.bind(console, 'connection err:'));
+db.once('open', function() {
```

Comment:
You should keep the style consistent and not use `function` and `=>` notation. I'd go with:

```
db.once('open', () => {
   ...
});
```
You can add an `eslint` rule for `=>`, and correct them automatically.
Same for semicolons.